pyopia/background.py

```diff
@@ -100,22 +100,27 @@ def correct_im_accurate(imbg, imraw):
     highlights if the background or raw images are very poorly obtained
 
     Args:
-      imbg (uint8)  : background averaged image
-      imraw (uint8) : raw image
+      imbg (uint8 or float64)  : background averaged image
+      imraw (uint8 or float64) : raw image
 
     Returns:
-      imc (uint8)   : corrected image
+      imc (uint8 or float64)   : corrected image, same type as input
     '''
 
     imc = np.float64(imraw) - np.float64(imbg)
-    imc[:, :, 0] += (255 / 2 - np.percentile(imc[:, :, 0], 50))
-    imc[:, :, 1] += (255 / 2 - np.percentile(imc[:, :, 1], 50))
-    imc[:, :, 2] += (255 / 2 - np.percentile(imc[:, :, 2], 50))
+    if imc.ndim == 3:
+        imc[:, :, 0] += (255 / 2 - np.percentile(imc[:, :, 0], 50))
+        imc[:, :, 1] += (255 / 2 - np.percentile(imc[:, :, 1], 50))
+        imc[:, :, 2] += (255 / 2 - np.percentile(imc[:, :, 2], 50))
+    else:
+        imc += (255 / 2 - np.percentile(imc, 50))
```

Comment:
Just a note, that if not using uint8 images, then the scaling applied here assumes that the value range of pixels in the image could span from 0-255 (i.e. I am not sure what might happen if images were floats with range 0-1)